File: src/main/java/com/theonlytails/ruby/registries/EntityTypeRegistry.java
package com.theonlytails.ruby.registries;

import com.theonlytails.ruby.RubyMod;
import com.theonlytails.ruby.entities.RubySheepEntity;
import net.minecraft.entity.EntityClassification;
import net.minecraft.entity.EntityType;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class EntityTypeRegistry {
    public static final DeferredRegister<EntityType<?>> ENTITY_TYPES = DeferredRegister.create(ForgeRegistries.ENTITIES, RubyMod.MOD_ID);

    public static final RegistryObject<EntityType<RubySheepEntity>> RUBY_SHEEP = ENTITY_TYPES.register("ruby_sheep", () -> EntityType.Builder.create(RubySheepEntity::new, EntityClassification.CREATURE).size(0.625f, 1.25f)
            .build(new ResourceLocation(RubyMod.MOD_ID, "ruby_sheep").toString())
    );
}

File: src/main/kotlin/com/github/theonlytails/rubymod/registries/EntityTypeRegistry.kt
package com.github.theonlytails.rubymod.registries

import com.github.theonlytails.rubymod.RubyMod
import com.github.theonlytails.rubymod.entities.RubySheepEntity
import net.minecraft.entity.EntityClassification
import net.minecraft.entity.EntityType
import net.minecraft.util.ResourceLocation
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

/**
 * Registers custom entities, i.e. mobs.
 *
 * @author TheOnlyTails
 */
object EntityTypeRegistry {
	val ENTITY_TYPES =
		KDeferredRegister(ForgeRegistries.ENTITIES, RubyMod.MOD_ID)

	val RUBY_SHEEP: EntityType<RubySheepEntity> by ENTITY_TYPES.register("ruby_sheep") {
		EntityType.Builder.create(::RubySheepEntity,
			EntityClassification.CREATURE).size(0.625f, 1.25f)
			.build(ResourceLocation(RubyMod.MOD_ID, "ruby_sheep").toString())
	}
}